### PR TITLE
feat: delegate, undelegate and withdraw transactions

### PR DESCRIPTION
**Pull request by elmutt**
Delegate, Undelegate and Withdraw transactions.

TODO later: dry up the chain adapter

### CODE REVIEW

File: packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts [hunk 1]
```diff
@@ -163,6 +163,289 @@ export class ChainAdapter
...
+      return { txToSign }
+    } catch (err) {
+      return ErrorHandler(err)
+    }
+  }
+
+  async buildUndelegateTransaction(
+    tx: chainAdapters.BuildUndelegateTxInput<ChainTypes.Cosmos>
+  ): Promise<{ txToSign: CosmosSignTx }> {
+    const {
```
Comment (cjthompson) [inline packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts:241]: wrap in `try/catch`

**Review by cjthompson — APPROVED** [review]
approved with comments

File: packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts (hunk 1 above)
Comment (elmutt) [reply packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts:241]: good catch

File: packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts [hunk 2]
```diff
@@ -163,6 +163,289 @@ export class ChainAdapter
...
+      account_number: account.chainSpecific.accountNumber,
+      sequence: account.chainSpecific.sequence
+    }
+    return { txToSign }
+  }
+
+  async buildClaimRewardsTransaction(
+    tx: chainAdapters.BuildClaimRewardsTxInput<ChainTypes.Cosmos>
+  ): Promise<{ txToSign: CosmosSignTx }> {
+    const {
```
[the remark this answers is not in the document]
Comment (elmutt) [reply packages/chain-adapters/src/cosmossdk/cosmos/CosmosChainAdapter.ts:308]: good catch

**Pull request merged**